### ISSUE TITLE
PaymentSheet with custom API client doesn't confirm payment with it

### ISSUE THREAD

**Issue by ricocrescenzio95** [stripe/stripe-ios#1898]
## Summary
In our app we had to use two different publishable keys, based on different flows.
When using a `PaymentSheet`, we need to customize its configuration to use the proper api client (initialized with the publishable key). 
`PaymentSheet` ignores the api client configuration when internally it uses the `STPPaymentHandler`; this causes the payment to finish with error 

```
Error Domain=com.stripe.lib Code=50 "There was an unexpected error -- try again in a few seconds" UserInfo={NSLocalizedDescription=There was an unexpected error -- try again in a few seconds, com.stripe.lib:StripeErrorTypeKey=invalid_request_error, com.stripe.lib:StripeErrorCodeKey=resource_missing, com.stripe.lib:ErrorParameterKey=intent, com.stripe.lib:ErrorMessageKey=No such payment_intent: 'pi_xxxxxxxxxxxxxxxxxxxxx'}
```

## Code to reproduce
```swift
let client = STPAPIClient(publishableKey: "a different publishable key")
var configuration = PaymentSheet.Configuration()
configuration.apiClient = client // setting my custom client to this configuration
configuration.customer = .init(id: "cus_xxxxx", ephemeralKeySecret: "ep_secret_xxxx")
paymentSheet = PaymentSheet(paymentIntentClientSecret: "pi_xxxxxx", configuration: configuration)
```

To fix this I have to explicit the client to the `STPPaymentHandler` shared instance too.

```swift
let client = STPAPIClient(publishableKey: "a different publishable key")
STPPaymentHandler.shared().apiClient = client
var configuration = PaymentSheet.Configuration()
configuration.apiClient = client
configuration.customer = .init(id: "cus_xxxxx", ephemeralKeySecret: "ep_secret_xxxx")
paymentSheet = PaymentSheet(paymentIntentClientSecret: "pi_xxxxxx", configuration: configuration)
```

## iOS version
15.0

## Installation method
Cocoapods

## SDK version
21.6.0

**Comment by yuki-stripe** [comment]
Thank you for raising this issue @ricocrescenzio95 - this will be fixed in the next release.

**Comment by davidme-stripe** [comment]
This was fixed in 21.10.0.
